web-client/integration-tests/permissions.test.js

```diff
@@ -0,0 +1,209 @@
+  expect(test.getState('caseDetail.docketEntries.0.isDraft')).toBeUndefined();
+  expect(test.getState('caseDetail.docketEntries.0.judge')).toBeUndefined();
+  expect(
+    test.getState('caseDetail.docketEntries.0.judgeUserId'),
+  ).toBeUndefined();
+  expect(test.getState('caseDetail.docketEntries.0.pending')).toBeUndefined();
+  expect(test.getState('caseDetail.docketEntries.0.qcAt')).toBeUndefined();
+  expect(
+    test.getState('caseDetail.docketEntries.0.qcByUserId'),
+  ).toBeUndefined();
+  expect(test.getState('caseDetail.docketEntries.0.signedAt')).toBeUndefined();
+  expect(
+    test.getState('caseDetail.docketEntries.0.signedByUserId'),
+  ).toBeUndefined();
+  expect(
+    test.getState('caseDetail.docketEntries.0.signedJudgeName'),
+  ).toBeUndefined();
+  expect(
+    test.getState('caseDetail.docketEntries.0.signedJudgeUserId'),
+  ).toBeUndefined();
+  expect(
+    test.getState('caseDetail.docketEntries.0.strickenBy'),
+  ).toBeUndefined();
+  expect(
+    test.getState('caseDetail.docketEntries.0.strickenByUserId'),
+  ).toBeUndefined();
+  expect(test.getState('caseDetail.docketEntries.0.workItem')).toBeUndefined();
+};
+
+const stinVisible = () => {
+  expect(
+    some(test.getState('caseDetail.docketEntries'), { eventCode: 'STIN' }),
+  ).toBe(true);
+};
+
+const stinBlocked = () => {
+  expect(
+    some(test.getState('caseDetail.docketEntries'), { eventCode: 'STIN' }),
+  ).toBe(false);
+};
+
+describe('Case permissions test', () => {
+  beforeEach(() => {
+    jest.setTimeout(30000);
+    global.window = {
+      ...global.window,
+      localStorage: {
+        removeItem: () => null,
+        setItem: () => null,
+      },
+    };
+  });
+
+  loginAs(test, 'petitioner@example.com');
+  petitionerCreatesNewCase(test, fakeFile);
+  petitionerViewsCaseDetail(test);
+  it('Petitioner views case detail', async () => {
+    test.setState('caseDetail', {});
+    await test.runSequence('gotoCaseDetailSequence', {
+      docketNumber: test.docketNumber,
+    });
+
+    publicFieldsVisible();
+    associatedFieldsVisible();
+    internalFieldsBlocked();
+    stinBlocked();
+
+    expect(
+      some(test.getState('caseDetail.docketEntries'), { eventCode: 'STIN' }),
+    ).toBe(false);
+  });
+
+  loginAs(test, 'irsSuperuser@example.com');
+  it('IRS Super User views case detail', async () => {
+    test.setState('caseDetail', {});
+    await test.runSequence('gotoCaseDetailSequence', {
+      docketNumber: test.docketNumber,
+    });
+
+    publicFieldsVisible();
+    associatedFieldsVisible();
+    internalFieldsBlocked();
+    stinVisible();
+  });
+
+  loginAs(test, 'privatePractitioner@example.com');
+  it('Unassociated private practitioner views case detail', async () => {
+    test.setState('caseDetail', {});
+    await test.runSequence('gotoCaseDetailSequence', {
+      docketNumber: test.docketNumber,
+    });
+
+    publicFieldsVisible();
+    // associatedFieldsBlocked();
```

Comment:
I missed this in my previous passthrough, @ericsorenson, do you know why it's commented out?

Reply:
The story was in progress.